graphql/src/store/query.rs

```diff
@@ -265,6 +407,13 @@ fn build_filter_from_object(
             use self::sast::FilterOp::*;
             let (field_name, op) = sast::parse_field_as_filter(key);
 
+            let actual_entity = match is_connection_type(entity.name()) {
+                false => entity,
+                true => schema
+                    .object_or_interface(&entity.name().replace("Connection", ""))
```

Comment:
You should define a helper `connection_base_type(&str) -> &str` that does this without allocation. It would also be nicer to have a helper on `ApiSchema` that does this entire `match` statement.